src/Action.hs

```diff
@@ -76,6 +82,8 @@ instance ActionUserHandler Action where
         put $ UserLoggedIn username "session"
         persistent $ loginUser username
 
+    userState = get
+
     logout = do
         gets _username >>= persistent . logoutUser
```

Comment:
This is not related to your commit but `gets _username` can be replaced by `use username`.